ambry-store/src/main/java/com.github.ambry.store/LogSegmentNameHelper.java

```diff
@@ -0,0 +1,162 @@
+  static int hashcode(String name) {
+    return name.hashCode();
+  }
+
+  /**
+   * @param name the name of the log segment.
+   * @return the relative position of the log segment.
+   */
+  static long getPosition(String name) {
+    if (name.isEmpty()) {
+      throw new IllegalArgumentException("Name provided cannot be empty");
+    }
+    return Long.parseLong(name.substring(0, name.indexOf(BlobStore.SEPARATOR)));
+  }
+
+  /**
+   * @param name the name of the log segment.
+   * @return the generation number of the log segment.
+   */
+  static long getGeneration(String name) {
+    if (name.isEmpty()) {
+      throw new IllegalArgumentException("Name provided cannot be empty");
+    }
+    return Long.parseLong(name.substring(name.indexOf(BlobStore.SEPARATOR) + 1, name.length()));
+  }
+
+  /**
+   * @param pos the relative position of the log segment.
+   * @param gen the generation of the log segment.
+   * @return the name of a log segment with position {@code pos} and generation number {@code gen}.
+   */
+  static String getName(long pos, long gen) {
+    return pos + BlobStore.SEPARATOR + gen;
+  }
+
+  /**
+   * @param name the name of the log segment.
+   * @return what should be the name of the log segment that is exactly one position higher than {@code name}.
+   */
+  static String getNextPositionName(String name) {
+    long pos = getPosition(name);
+    long gen = getGeneration(name);
+    return getName(pos + 1, gen);
+  }
+
+  /**
+   * @param name the name of the log segment.
+   * @return what should be the name of the log segment that is exactly one generation higher than {@code name}.
+   */
+  static String getNextGenerationName(String name) {
+    long pos = getPosition(name);
+    long gen = getGeneration(name);
+    return getName(pos, gen + 1);
+  }
+
+  /**
+   * @param numSegments the total number of segments in the log.
+   * @return what should be the name of the first segment.
+   */
+  static String generateFirstSegmentName(long numSegments) {
+    if (numSegments <= 0) {
+      throw new IllegalArgumentException("Number of segments <=0");
+    }
+    if (numSegments == 1) {
+      return "";
+    }
+    return getName(0, 0);
+  }
+
+  /**
+   * @param name the name of the log segment.
+   * @return the name of the file that backs the log segment.
+   */
+  static String nameToFilename(String name) {
+    if (name.isEmpty()) {
+      return SINGLE_SEGMENT_LOG_FILE_NAME;
+    }
+    return name + SUFFIX;
+  }
+
+  /**
+   * @param filename the name of the file that backs the log segment.
+   * @return the name of the log segment.
+   */
+  static String nameFromFilename(String filename) {
+    if (filename.equals(SINGLE_SEGMENT_LOG_FILE_NAME)) {
+      return "";
+    }
+    if (!filename.endsWith(SUFFIX)) {
+      throw new IllegalArgumentException("The filename of the log segment does not end with [" + SUFFIX + "]");
+    }
+    return filename.substring(0, filename.length() - SUFFIX.length());
+  }
+}
```

Comment:
version in the header is better as long as we recognize that log name format is part of the version and if that is changed, the version needs to be changed as well.